db/model/sample.js

```diff
@@ -170,49 +170,84 @@ module.exports = function sample(seq, dataTypes) {
        *  encountered while performing the sample upsert operation itself.
        */
       upsertByName(toUpsert, isBulk) {
-        let sampleExists = true;
-        return new seq.Promise((resolve, reject) => {
-          // get sample by name
-          Sample.findOne({
-            where: {
-              name: {
-                $iLike: toUpsert.name,
+        if (config.optimizeUpsert === 'true' ||
```

Comment:
We should put the truthiness check (`(config.optimizeUpsert === 'true' || config.optimizeUpsert === true)`) in config.js so that all the other places that might need to check the variable can just do the simpler easier-to-read `if (config.optimizeUpsert)`